src/otel-collector/otelcol-config.yml

```diff
@@ -153,6 +153,28 @@ processors:
           # could be removed when https://github.com/vercel/next.js/pull/64852 is fixed upstream
           - replace_pattern(name, "\\?.*", "")
           - replace_match(name, "GET /api/products/*", "GET /api/products/{productId}")
+          # Workaround for https://github.com/opensearch-project/data-prepper/issues/5616
+          # Flatten nested dotted attribute keys to prevent OpenSearch mapping conflicts
+          # Copy nested values to flattened keys, then delete nested keys
+          - set(attributes["db_system_name"], attributes["db.system.name"]) where attributes["db.system.name"] != nil
+          - delete_key(attributes, "db.system.name") where attributes["db.system.name"] != nil
+          - set(attributes["code_function_name"], attributes["code.function.name"]) where attributes["code.function.name"] != nil
+          - delete_key(attributes, "code.function.name") where attributes["code.function.name"] != nil
+          - set(attributes["user_agent_original"], attributes["user_agent.original"]) where attributes["user_agent.original"] != nil
+          - delete_key(attributes, "user_agent.original") where attributes["user_agent.original"] != nil
+          - set(attributes["upstream_cluster_name"], attributes["upstream_cluster.name"]) where attributes["upstream_cluster.name"] != nil
+          - delete_key(attributes, "upstream_cluster.name") where attributes["upstream_cluster.name"] != nil
+          - set(attributes["error_type"], attributes["error.type"]) where attributes["error.type"] != nil
+          - delete_key(attributes, "error.type") where attributes["error.type"] != nil
+          - set(attributes["app_ads_contextKeys"], attributes["app.ads.contextKeys"]) where attributes["app.ads.contextKeys"] != nil
+          - delete_key(attributes, "app.ads.contextKeys") where attributes["app.ads.contextKeys"] != nil
+    log_statements:
+      - context: log
+        statements:
+          # Handle structured log bodies (maps/objects) that cause parse errors in OpenSearch
+          # Replace with a simple string and preserve structure in attributes
+          - set(attributes["original_body"], body) where IsMap(body)
```

Comment:
I think we should be body stringified rather than moving it to another field. Myrationale is we can use PPL json parse commands for this. `- set(body, ConvertMapToString(body)) where IsMap(body)`